core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -690,12 +691,13 @@ private[spark] class TaskSetManager(
   }
 
   /**
-   * Check whether has enough quota to fetch the result with `size` bytes
+   * Check whether has enough quota to fetch the result with `size` bytes.
+   * This check does not apply to shuffle map tasks as they return map status and metrics updates.
```

Comment:
`..., which will not be held by the driver and will be thrown away when the driver finishes processing it.`